@xen-orchestra/fs/src/abstract.js

```diff
@@ -39,15 +39,22 @@ class PrefixWrapper {
   }
 
   // necessary to remove the prefix from the path with `prependDir` option
-  async list(dir, opts) {
-    const entries = await this._handler.list(this._resolve(dir), opts)
-    if (opts != null && opts.prependDir) {
-      const n = this._prefix.length
-      entries.forEach((entry, i, entries) => {
-        entries[i] = entry.slice(n)
-      })
+  async list(dir, { ignoreEnoentError = false, ...opts } = {}) {
```

Comment:
While you're at it, you can destructure `prependDir` as well.

It's a local option that shouldn't be passed further down.